splash/qtrender_lua.py

```diff
@@ -456,10 +456,13 @@ def get_viewport_size(self):
     @command()
     def set_viewport_size(self, width, height):
         self.tab.set_viewport('%dx%d' % (width, height))
+        process_pending_events()
 
     @command(multiple_return_values=True)
     def set_viewport_full(self):
-        return list(self.tab.set_viewport('full'))
+        size = self.tab.set_viewport('full')
+        process_pending_events()
```

Comment:
Is it different from e.g. waiting for 0 seconds?
I'm asking because the question is what may happen during processEvents call. For example, can the page redirect to a different URL?